### DISCUSSION TITLE
Twinning

### DISCUSSION THREAD

**Discussion by Debo-rah** [mtex-toolbox/mtex#1302] [Ask Anything]
Hi Mtex community,

I have a general question on twinning and I think my question is more on the side of understanding how it is defined with MTEX. There are different angles and axes and planes given in literature and I dont know which of these definitions is used in the ways of plotting twins I found in the MTEX Documentation.

I am looking at Calcite. What I get from literature is the twinning plane and the angles between the c-axis. E.g.
Twinning at the 10-14 planes with an angle between the c-axes in the original and the twin of 90°. 
Or Twinning at the 01-12 planes with an angle between the c-axes of 53,74°.

In Mtex, I found the approach to go via the angle, but I dont understand for which angle I have to look. 

1.) The misorientation of the grainboundary , e.g. 
plotAngleDistribution(gB.misorientation) 
would not fit my angles between the c-axes right? 

2.) Misorientation of all the Ebsd data 
mori = ebsd('C').mis2mean;
plotAngleDistribution(mori)

I see different angles than with the first approach but still dont see dominant twinning angles.  So both approaches didnt help me. 

3.) In the forum I found someone trying it with
twinning = orientation.map(Miller({1,0,-1,-3},CS) ,Miller({0,0,0,1},CS,...
Miller({1,-2,1,0},CS('Calcite')), Miller({2,-1,-1,0},CS)) 
as a condition where, pairs of parallel planes are defined? Unfortunately I dont understand how to get the indices of parallel planes for my system. 

I would like to plot the twinboundary in my EBSD map, so in the documentation I found this:

%104 twin
twin104 = orientation('axis',Miller(1,0,-1,4,cs),'uvw','angle',90*degree,cs,cs);
isTwinning104 = angle(gB.misorientation,twin104) < 5*degree;
twinBoundary104 = gB(isTwinning104);

%012 twin
twin012 = orientation('axis',Miller(0,1,-1,2,cs),'uvw','angle',26*degree,cs,cs);
isTwinning012 = angle(gB.misorientation,twin012) < 5*degree;
twinBoundary012 = gB(isTwinning012);

With this approach I didnt get any twin boundaries in my system, even though from the polefigrues its quite clear its twinnning. However I think also here I have to find another angle, since 90deg is the angle between c-axes, but I think MTEX wants another angle. I dont know which one?  Is it just half of that angle, so the angle between the twinplane and the c-axis? And does it matter which angle I chose to define a grainboundary?

I attach one of my polefigures where I would say I see a 012 twin...
![grain_02_subgrain_twin_PDF](https://user-images.githubusercontent.com/72927930/145382907-669f3de3-553e-4f98-8277-6616e5f9729b.png)

Thanks a lot for your help !

**Comment by kilir** [comment]
Hi,
you have a syntax error when defining your misorientations. The axis should be a 'uvw' or 'UVTW'. However, you close your brackets prematurely.
Cheers,
Rüdiger

**Reply by Debo-rah** [reply]
Hi Rüdiger, 

thanks for helping! I'll try with uvtw again. Unfortunatley I dont find the brackets problem. Can you tell me where?

twin012 = orientation('axis',Miller(0,1,-1,2,cs),'uvtw','angle',56*degree,cs,cs);
isTwinning012 = angle(gB.misorientation,twin012) < 5degree;
twinBoundary012 = gB(isTwinning012);

Is it right like this? I am a bit confused by the word "axis" cause the 01-12 plane is actually the twinning plane. Is it the same as axis? And the angle, should it be the angle between the c-axes or the angle between the twin plane and the c-axes?

Thanks a lot for your help!
Deborah

**Reply by kilir** [reply]
Hi,
you need to define your twinning in terms of the misorientation associated with it and if you choose to define those by axis - angle, the axis is the misorientation axis and the angle is the misorientation angle associated with your twinning relationship. There are other ways to define those orientation relations e.g. `orientation.map()`. Depending on your twinning relationship, some ways to define are more convenient than others. If you just have a composition plane and a direction, it's probably easiest to use `orientation.map()`.

 My reference to your syntax was that you use an axis but you have a 'uvw' somewhere, where it does not do anything - not inside the Miller() ). However, if your have a misorientations axis which is defined by a plane normal, it should indeed be only Miller() without the 'uvw'.

Also take care, if you take those twinning relations from the literature, you need to be a bit careful to a) what uni cell parameters it was referred to (crystallographic vs morphological). E.g. an e-twin would be 180° around (01-18) in the crystallographic definition of the unitcell, However, one might also read that it is 180° around (01-12) which is the same but given in the morphological description of the unitcell. The above twinning misorientation in mtex could be:
```
cs = crystalSymmetry('-3m',[4.99 4.99 17.06],'X||a*')
etwin = orientation.byAxisAngle(Miller(0,1,-1,8,cs),180*degree)
```
Hope that helps,
Rüdiger

**Reply by Debo-rah** [reply]
hi Rüdiger,
thanks a lot for your help! 
I was indeed speaking about the crystallographic description of the unit cell.
The twin planes are then c={0001}, r={10-14}, e={01-18}, f={01-12} 
Since the twinplane is a mirror plane, can I define the twin alsways as axis angle pair always with the corresponding Miller indices of the plane as "axis" and 180° of misorientation? I am not sure about this since it is the hexagonal system and directions are a bit complicated there..? 

```
 %c-twin
ctwin = orientation('axis',Miller(0,0,0,1,cs),'angle',180*degree,cs,cs);

%e-twin
etwin = orientation('axis',Miller(0,1,-1,8,cs),'angle',180*degree,cs,cs);

%r- twin
rtwin = orientation('axis',Miller(1,0,-1,4,cs),'angle',180*degree,cs,cs);

%f- twin
ftwin = orientation('axis',Miller(0,1,-1,2,cs),'angle',180*degree,cs,cs);
```

Have a nice weekend!
Deborah
